File: week6/31861/burningfalls/31861.cpp
#include <bits/stdc++.h>
using namespace std;

#define FASTIO cin.tie(0); ios_base::sync_with_stdio(0);
#define ll long long
#define MOD 1000000007

int main() {
    FASTIO;
    int N, M;
    cin >> N >> M;
    vector<vector<ll>> dp(M + 1, vector<ll>(26, 1));
    for(int i = 2; i <= M; i++) {
        for(int j = 0; j < 26; j++) {
            ll sum = 0;
            for(int k = 0; k < 26; k++) {
                if (abs(j - k) >= N) {
                    sum = (sum + dp[i - 1][k]) % MOD;
                }
            }
            dp[i][j] = sum;
        }
    }
    
    ll ans = 0;
    for(int i = 0; i < 26; i++) {
        ans = (ans + dp[M][i]) % MOD;
    }
    cout << ans << "\n";
    
    return 0;
}

File: week6/31861/hjk0761/31861.py
n, m = map(int, input().split())

dp = [[0 for _ in range(26)] for _ in range(m)]

dp[0] = [1 for _ in range(26)]

for i in range(1, m):
    for j in range(26):
        for k in range(26):
            if abs(j - k) >= n:
                dp[i][j] += dp[i-1][k]
        dp[i][j] %= 1000000007
        
print(sum(dp[-1])%1000000007)